subsys/caf/modules/ble_adv.c

```diff
@@ -244,69 +295,48 @@ static int ble_adv_start_undirected(const bt_addr_le_t *bond_addr,
 		adv_param.interval_max = CONFIG_CAF_BLE_ADV_SLOW_INT_MAX;
 	}
 
-	if (IS_ENABLED(CONFIG_CAF_BLE_ADV_FILTER_ACCEPT_LIST)) {
-		int err = bt_le_filter_accept_list_clear();
+	adv_param.id = cur_identity;
 
-		if (err) {
-			LOG_ERR("Cannot clear filter accept list (err: %d)", err);
-			return err;
-		}
+	int allowed_cnt = setup_accept_list(cur_identity);
 
-		if (bt_addr_le_cmp(bond_addr, BT_ADDR_LE_ANY)) {
-			adv_param.options |= BT_LE_ADV_OPT_FILTER_SCAN_REQ;
-			adv_param.options |= BT_LE_ADV_OPT_FILTER_CONN;
-			err = bt_le_filter_accept_list_add(bond_addr);
-		}
-
-		if (err) {
-			LOG_ERR("Cannot add peer to filter accept list (err: %d)", err);
-			return err;
-		}
+	if (allowed_cnt < 0) {
+		return allowed_cnt;
+	} else if (allowed_cnt > 0) {
+		adv_param.options |= BT_LE_ADV_OPT_FILTER_SCAN_REQ;
+		adv_param.options |= BT_LE_ADV_OPT_FILTER_CONN;
 	}
 
-	adv_param.id = cur_identity;
-
 	return update_undirected_advertising(&adv_param, false);
 }
 
 static int ble_adv_start(bool can_fast_adv)
 {
 	bool fast_adv = IS_ENABLED(CONFIG_CAF_BLE_ADV_FAST_ADV) && can_fast_adv;
 
-	struct bond_find_data bond_find_data = {
-		.peer_id = 0,
-		.peer_count = 0,
-	};
-	bt_addr_le_copy(&bond_find_data.peer_address, BT_ADDR_LE_ANY);
-	bt_foreach_bond(cur_identity, bond_find, &bond_find_data);
-
 	int err = ble_adv_stop();
 	if (err) {
 		LOG_ERR("Cannot stop advertising (err %d)", err);
 		goto error;
 	}
 
-	struct bt_conn *conn = NULL;
-
-	bt_conn_foreach(BT_CONN_TYPE_LE, conn_find_cb, &conn);
-	if (conn) {
+	if (conn_get()) {
 		LOG_INF("Already connected, do not advertise");
 		return 0;
 	}
 
 	bool direct = false;
 
-	if (bond_find_data.peer_id < bond_find_data.peer_count) {
-		if (IS_ENABLED(CONFIG_CAF_BLE_ADV_DIRECT_ADV)) {
-			/* Direct advertising only to peer without RPA. */
-			direct = (peer_is_rpa[cur_identity] != PEER_RPA_YES);
-		}
+	/* Use direct advertising only if the local identity in use has exactly one bond. */
```

Comment:
why do we limit the direct adv in such a way?

Reply:
nRF Desktop by default supports one bond per Bluetooth local identity (so in most cases the condition in comment is true anyway). If we have more than one bond, we would advertise directly always towards the first bond (which would be confusing). I think it's better to use undirected advertising in that case.

Reply:
> If we have more than one bond, we would advertise directly always towards the first bond (which would be confusing).

Yes, this would prevent other bonded peers from connecting. In such a case, it would make more sense to advertise directly to the bond that belongs to the most recent peer (in terms of connection) and then switch to the undirected advertising after a short timeout.

Reply:
For now, we stay in directed advertising and not switch to the undirected advertising automatically. Still, maybe we should do it instead of checking if our peer uses RPA.